File: CSharpFundamentals/CSharpOOPbasics/EncapsulationLab/SortPerson/Person.cs
namespace PersonsInfo
{
    public class Person
    {
        private string firstName;
        private string lastName;
        private int age;

        public Person(string firstName, string lastName, int age)
        {
            this.firstName = firstName;
            this.lastName = lastName;
            this.age = age;
        }

        public int Age
        {
            get { return age; }
        }

        public string LastName
        {
            get { return lastName; }
        }

        public string FirstName
        {
            get { return firstName; }
        }

        public override string ToString()
        {
            return $"{this.FirstName} {this.LastName} is {this.Age} years old.";
        }
    }
}

File: JavaFundamentals/JavaAdvanced/DefiningClasses_Ex/src/FamilyTree/Person.java
package FamilyTree;

import java.util.ArrayList;
import java.util.List;

public class Person {
    private static int idCounter = 1;
    private int id;
    private String firstName = "kur";
    private String lastName = "kur";
    private String birthday = "kur";
    private List<Person> parents;
    private List<Person> children;

    public Person() {
        this.id = idCounter++;
        parents = new ArrayList<>();
        children = new ArrayList<>();
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getBirthday() {
        return birthday;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public void addParent(Person person) {
        this.parents.add(person);
    }

    public void addChld(Person person) {
        this.children.add(person);
    }

    public int getId() {
        return id;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append(String.format("%s %s %s",
                this.firstName, this.lastName, this.birthday))
        .append(System.lineSeparator());
        sb.append("Parents:").append(System.lineSeparator());
        for (Person parent : this.parents) {
            sb.append(String.format("%s %s %s",
                    parent.firstName, parent.lastName, parent.birthday));
        }
        sb.append("Children:").append(System.lineSeparator());
        for (Person child : this.children) {
            sb.append(String.format("%s %s %s",
                    child.firstName, child.lastName, child.birthday));
        }

        return sb.toString();
    }
}
